### PR TITLE
INTYGFV-14330: Added possibility for components to dispatch client validation errors

### PR DESCRIPTION
**Pull request by mhornfeldt**

### CODE REVIEW

**Comment by mhornfeldt** [comment]
PR is draft should not be reviewed

File: packages/common/src/components/Inputs/DatePickerCustom/DatePickerCustom.tsx
Comment (mww-sopra-steria) [inline packages/common/src/components/Inputs/DatePickerCustom/DatePickerCustom.tsx:9]: We have to avoid dependencies from common to webcert. See if you can move something more to common so we can get rid this dependency to webcert.

File: packages/webcert/src/store/certificate/certificateSelectors.ts
Comment (mww-sopra-steria) [inline packages/webcert/src/store/certificate/certificateSelectors.ts:155]: For performance reason memoize the clientValidation errors so that all the filtering logic doesn't have to be done if there is no change.

Here is a link: https://react-redux.js.org/api/hooks#using-memoizing-selectors

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Great job! A couple of comments that you should take a look at before merging.

File: packages/webcert/src/store/certificate/certificateSelectors.ts
Comment (mhornfeldt) [reply packages/webcert/src/store/certificate/certificateSelectors.ts:155]: Is it possible to do this when we need attributes from the component to do the filtering?

Comment (mhornfeldt) [reply packages/webcert/src/store/certificate/certificateSelectors.ts:155]: @mww-sopra-steria

Comment (mhornfeldt) [reply packages/webcert/src/store/certificate/certificateSelectors.ts:155]: @mww-sopra-steria Is it really possible to do use memoize when we need attributes from the component to do the filtering? If we don't have the questionId and field we can't filter out the validationMessages for the specific component

Comment (mww-sopra-steria) [reply packages/webcert/src/store/certificate/certificateSelectors.ts:155]: I turn that question around a little. Have you found any documentation or tried and found that it won't work?

Comment (mhornfeldt) [reply packages/webcert/src/store/certificate/certificateSelectors.ts:155]: @mww-sopra-steria https://stackoverflow.com/questions/40291084/use-reselect-selector-with-parameters

Comment (mww-sopra-steria) [reply packages/webcert/src/store/certificate/certificateSelectors.ts:155]: It seems like there are ways to implement it, but I don't find it so intuitive or easy. I suggest you merge the PR and then I will analyze what the possible impact the current implementation has. If it is none to minimal, we can be satisfied with the current selector. If not it will be easier to experiment with the client by having all the changes integrated in the master branch.

**Review by mww-sopra-steria — APPROVED** [review]
See my last comment relating to the selector and memoization.

**Pull request merged**
